src/snowflake/cli/_plugins/spcs/services/commands.py

```diff
@@ -221,20 +225,68 @@ def logs(
     num_lines: int = typer.Option(
         500, "--num-lines", help="Number of lines to retrieve."
     ),
+    previous_logs: bool = typer.Option(
+        False, "--previous-logs", "-p", help="Retrieve previous logs", is_flag=True
+    ),
+    since_timestamp: Optional[str] = typer.Option(
+        "", "--since", help="Timestamp to retrieve logs from"
+    ),
+    include_timestamps: bool = typer.Option(
+        False, "--include-timestamps", help="Include timestamps in logs", is_flag=True
+    ),
+    follow: bool = typer.Option(
+        False, "--follow", "-f", help="Continue polling for logs.", is_flag=True
+    ),
+    follow_interval: int = typer.Option(
+        2,
+        "--follow-interval",
+        help="Polling interval in seconds when using the --follow flag",
+    ),
```

Comment:
Why not recommend using `watch -n snow spcs service logs`?

Reply:
There are several reasons for this

1. `watch -n 2 snow spcs service logs` has some drawbacks, this call will get last 1000 log lines every time invoked.
2.  The functionality that we want to support is streaming (To append new logs to a current set of logs).
      - We offer customers to stream from a particular timestamp using --since flag
      - We offer deduplication logic
      - We are also going to perform load testing to understand backend api limits anc come up with a follow interval
      - We plan to implement exponential backoff
3.  Here is the design doc - https://docs.google.com/document/d/1hvGC97jfvhfcg6gyPnFUl5j2-Xc1MKWy6C7ftc5JZ1c/edit?tab=t.0#heading=h.an8exbgprj8y
4.  This is also similar to what the events does here currently - 
https://github.com/snowflakedb/snowflake-cli/blob/dfefe77e68ee88a4f529d9eef6a6f6c1bf71b09c/src/snowflake/cli/_plugins/nativeapp/entities/application.py#L816